### PR TITLE
Feat/left right blocks

### PR DESCRIPTION
**Pull request by aleksandrgordeev**
# Context

Link to [Notion ticket](https://www.notion.so/cloudcannon/Left-Right-Blocks-259af9e8899346f8a5f53cfe006320d4?pvs=4)

# Additional information

Implements Left/Right Blocks. 

# Currently not implemented

A generic button component is still in development so I used a pure HTML button without any styles for now.

# Side changes

1. Set a max-width for stacked labelled icons. This helps to add any amount of icons in a row. Iconst not fitting in one row will be transferred to a second row.
2. Add an uploads folder to build for the user's uploaded assets

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component in Bookshop browser
- The site in CloudCannon ([link here](https://sparkling-fall.cloudvent.net/))

# Before merge (PR owner)

- [ ] Delete the test site in CloudCannon
- [ ] For "generic" components **only**: remove `- content_blocks` from `structures` in the YAML file (this was added for testing).

### CODE REVIEW

**Comment by AlyshaNolan** [comment]
I started reviewing this and have found a couple things to potentially change:
- Rename key from data -> content (to match other components)
- Description something like -> Content that can be arranged left-right or right-left (rather than mostly repeating the label)
- Make a generic component left right that can take any blocks either side and make specific variants of it (3 different components rather than 1 that handles everything)
- change assign -> assign_local
- Should the content be markdown or just straight text?
- Pass content to text-block (without using text block for the input)

**Review by AlyshaNolan — CHANGES_REQUESTED** [review]
Please see my comment 😊

File: component-library/components/sections/testimonial/testimonial.bookshop.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    options:
+      values:
+        - Left
+        - Right  
+  background_variant:
+    type: select
+    options:
+      values:
+        - 'Dark'
+        - 'Light'
```
Comment (AlyshaNolan) [inline component-library/components/sections/testimonial/testimonial.bookshop.yml]:
```suggestion
```

File: component-library/components/sections/featured-icons/featured-icons.bookshop.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    options:
+      values:
+        - Left
+        - Right  
+  background_variant:
+    type: select
+    options:
+      values:
+        - 'Dark'
+        - 'Light'
```
Comment (AlyshaNolan) [inline component-library/components/sections/featured-icons/featured-icons.bookshop.yml]:
```suggestion
```

Comment (AlyshaNolan) [reply component-library/components/sections/featured-icons/featured-icons.bookshop.yml]: This is in the global config

File: component-library/components/sections/featured-icons/featured-icons.eleventy.liquid [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          hero_icon_name: {{ icon_data.hero_icon_name }}
+          icon_size: large
+          icon_type: outline
+          rounded_border: true
+          color: var(--main-text-color)
+    {% endcapture %}
+    {%  assign block_icons = block_icons | append: labelled-icon %}    
+  {% else %}
+    {% comment %}
+      TBD after adding custom icons support to labelled-icon
```
Comment (AlyshaNolan) [inline component-library/components/sections/featured-icons/featured-icons.eleventy.liquid]:
```suggestion
      TODO: after adding custom icons support to labelled-icon
```

**Review by AlyshaNolan — CHANGES_REQUESTED** [review]
A couple things from me:

- [x]  I think structures can be used to improve the generic component slightly - I can wheel over to your desk tomorrow to work through this one with you (if you like) as it is a lil complicated
- [x]  I think the images should be moved from under `style` to under `content`
- [x] Can `orientation` be made more descriptive to what its for - e.g. `image_alignment` (this is how other components such as hero implement this)
- [x] Background variants have changed - now `light` -> `primary` and `dark` -> `secondary`
- [x]  Can we name the blocks left-right-block--{{ component name e.g. testimonial }}

**Review by AlyshaNolan — CHANGES_REQUESTED** [review]
- [x] Can the components use the heading subcomponent - then the capture for the heading could be removed too
- [x] Label should be Left/Right - (name - e.g. Testimonial)
- [x] Image should be full height

File: component-library/components/sections/featured-icons/featured-icons.bookshop.yml (hunk 2 above)
Comment (aleksandrgordeev) [reply component-library/components/sections/featured-icons/featured-icons.bookshop.yml]: Done

File: component-library/components/sections/featured-icons/featured-icons.eleventy.liquid (hunk 3 above)
Comment (aleksandrgordeev) [reply component-library/components/sections/featured-icons/featured-icons.eleventy.liquid]: Thanks for the fix 👍

File: component-library/components/sections/testimonial/testimonial.bookshop.yml (hunk 1 above)
Comment (aleksandrgordeev) [reply component-library/components/sections/testimonial/testimonial.bookshop.yml]: Done

**Review by AlyshaNolan — APPROVED** [review]

**Comment by AlyshaNolan** [comment]
Approved with the assumption components will be removed from the page and uploads will be removed

**Pull request merged**